ios/engine/KMEI/KeymanEngine/Classes/Queries/Queries.swift

```diff
@@ -14,15 +14,18 @@ public class Queries {
     case networkError(Error)
     case noData
     case parsingError(Error)
+    case decodingError(String, String)
 
     var localizedDescription: String {
       switch self {
         case .noData:
-          return "no data returned"
-        case .networkError(let baseError):
-          return "network error occurred - \(String(describing: baseError))"
-        case .parsingError(let baseError):
-          return "failure occurred when parsing results - \(String(describing: baseError))"
+          return engineBundle.localizedString(forKey: "error-query-no-data", value: nil, table: nil)
+        case .networkError(_):
+          return engineBundle.localizedString(forKey: "error-query-general", value: nil, table: nil)
```

Comment:
Why is the key for network error a "general" one?

Reply:
https://github.com/keymanapp/keyman/blob/8b47386ca05a8e7116f1ddf579ee614920845c66/ios/engine/KMEI/KeymanEngine/en.lproj/Localizable.strings#L58-L59

I felt it wiser to go for a more general error for any alert messaging.  This case arises when the system reports network errors, so it's kind of hard for me to anticipate ways to provide meaningful error messages here... at least in ways that wouldn't seem like major technobabble to average users.

For programmers, there's an attached parameter (of type `Error`) that contains extra data that could be used in, say, Sentry reports.